src/Lecture3.hs

```diff
@@ -102,86 +101,104 @@ have a lawful 'Monoid' instance.
 
 -}
 
+-- NOTE (Student) It is easier to derive num as Gold behaves like a
+-- simple Int anyway
```

Comment:
In this case yes, but generally deriving `Num` for `Gold` doesn't make sense because there's no such things as gold multiplication. Num has too many methods and is generally considered a flawed abstraction.

Reply:
I'll keep in mind that in mind for the ```Num``` typeclass. I removed the derivation and used the ```Gold``` ```<>``` where necessary instead.